libopenage/renderer/opengl/geometry.cpp

```diff
@@ -51,6 +51,10 @@ void GlGeometry::update_verts_offset(std::vector<uint8_t> const &verts, size_t o
 	this->mesh->vertices.upload_data(verts.data(), offset, verts.size());
 }
 
+std::optional<GLuint> GlGeometry::get_vao_handle() const {
+	return this->mesh->vao.get_handle();
```

Comment:
This should check if `mesh` is non-empty, and only try to return the VAO handle then. Otherwise, return empty.